splunk/packages/edit-table/src/components/KVStoreUploader.jsx

```diff
@@ -114,9 +176,54 @@ export default function KVStoreUploader({
         }
     };
 
+    const runIncrementalUpload = async () => {
```

Comment:
maybe put these two runners in two separate files? (incremental.js and replace.js)?

Reply:
can't we just pass the result of `getFormattedData()` and `{ splunkApp, kvStore, backupErrorMsg, restoreErrorMsg }` as parameter and make it a promise. `showErrorMessage` and `showSuccessMessage` can act on the result of the promise

Reply:
You convinced me :) The only thing I think we can do is adding a batch function that is doing inserts and updates, like this

```
batchUpsertCollectionEntries(splunkApp,
                    collectionName,
                    data,
                    uploadErrorMsg);
```                    

(that function is then also calling `getIncrementalUploadData`)    

Then we can leave the callback and just write

```
await doKvStoreChanges(options, async () => await Promise.all(
                        batchUpsertCollectionEntries(
                            splunkApp,
                            collectionName,
                            data,
                            updateErrorMsg
                        )
                    ));
```

what do you think?